src/auto/injector.js

```diff
@@ -337,7 +337,53 @@ function annotate(fn, strictDi, name) {
  *
  * @returns {Array.<string>} The names of the services which the function requires.
  */
-
+/**
+ * @ngdoc method
+ * @name $injector#loadNewModules
+ *
+ * @description
+ *
+ * **This is an dangerous API, which you use at your own risk!**
+ *
+ * Add the specified modules to the current injector.
+ *
+ * This method will add each of the injectables to the injector and execute all of the run blocks
+ * for each module passed to the method.
+ * * The application developer is responsible for loading the code containing the modules.
+ * * Modules cannot be unloaded.
+ * * Previously compiled HTML will not be affected by newly loaded directives, filters and components.
+ *
+ * ## Example
+ *
+ * Here is an example of loading a bundle of modules, with a utility method called `getScript`:
+ *
+ * ```javascript
+ * app.service('AmdLoader', function ($injector, $log) {
+ *
+ *  return function(name, config) {
+ *    getScript(config).done(function () {
+ *      $log.info('Script loaded : ' + name);
+ *      onReady(name);
+ *    });
+ *
+ *    function onReady(name) {
+ *      var moduleName = name + '.html.bundle';
+ *      $log.debug('AMD >> Loading Angular module [' + moduleName + '] ...');
+ *
+ *      if (config.modules) {
+ *        $injector.loadNewModules(config.modules);
+ *        $log.debug('AMD >> Children modules loaded >>>');
+ *        $log.debug(config.modules);
+ *      }
+ *
+ *      $injector.loadNewModules([moduleName]);
+ *      $log.debug('AMD >> Angular module [' + moduleName + '] loaded.');
+ *    }
+ * });
```

Comment:
FWIW, I find the example unnecessarily complicated. I would expect something like:

```js
app.factory('loadModule', function($injector) {
  return function loadModule(moduleName, bundleUrl) {
    return getScript(bundleUrl).then(function() { $injector.loadNewModules([moduleName]); });
  };
})
```